src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -1010,39 +1006,34 @@ public virtual byte[] GetBytes(String s)
         // string range.
         //
         [Pure]
-        public byte[] GetBytes(string s, int index, int count)
+        public unsafe byte[] GetBytes(string s, int index, int count)
         {
             if (s == null)
-                throw new ArgumentNullException(nameof(s),
-                    Environment.GetResourceString("ArgumentNull_String"));
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.s, ExceptionResource.ArgumentNull_String);
             if (index < 0)
-                throw new ArgumentOutOfRangeException(nameof(index),
-                      Environment.GetResourceString("ArgumentOutOfRange_NeedNonNegNum"));
+                ThrowHelper.ThrowArgumentOutOfRange_IndexException();
             if (count < 0)
-                throw new ArgumentOutOfRangeException(nameof(count),
-                      Environment.GetResourceString("ArgumentOutOfRange_NeedNonNegNum"));
+                ThrowHelper.ThrowCountArgumentOutOfRange_NeedNonNegNumException();
             if (index > s.Length - count)
-                throw new ArgumentOutOfRangeException(nameof(index),
-                      Environment.GetResourceString("ArgumentOutOfRange_IndexCount"));
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.index, ExceptionResource.ArgumentOutOfRange_IndexCountBuffer);
             Contract.EndContractBlock();
 
-            unsafe
-            {
-                fixed (char* pChar = s)
-                {
-                    int byteCount = GetByteCount(pChar + index, count);
-                    if (byteCount == 0)
-                        return Array.Empty<byte>();
-
-                    byte[] bytes = new byte[byteCount];
+            byte[] bytes;
+            fixed (char* pChar = s) {
+                int byteCount = GetByteCount(pChar + index, count);
+                if (byteCount == 0) {
```

Comment:
> if (byteCount == 0) { [](start = 15, length = 22)

please try to use the coding style 

if ()
{
}
else
{
}

I know the old code was not doing that but any changed code we should use the right style